src/ion/liveranges.rs

```diff
@@ -1013,13 +1027,33 @@ impl<'a, F: Function> Env<'a, F> {
                                             operand.pos(),
                                         ),
                                     );
+                                } else if self.func.inst_clobbers(inst).contains(preg) {
+                                    let pos = ProgPoint::before(inst);
+
+                                    // We need to insert a clobber at
+                                    // the before-point to reserve the
+                                    // reg for the use too.
+                                    let range = CodeRange {
+                                        from: ProgPoint::before(inst),
+                                        to: ProgPoint::after(inst),
+                                    };
```

Comment:
Would it be worth adding a `CodeRange::whole_instruction` function for constructing this special case?

Reply:
The range is actually just over the before-point (`from` is inclusive and `to` is exclusive) but I've created `CodeRange::singleton(pos)` for this -- thanks!